arcgis-ios-sdk-samples/Layers/Export tiles/ExportTilesViewController.swift

```diff
@@ -17,154 +17,242 @@ import UIKit
+            let scale = 1e7
+            mapView.map?.minScale = scale
+            zoomToViewpoint(mapView: mapView, scale: scale)
         }
     }
     
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        
-        //add the source code button item to the right of navigation bar
-        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["ExportTilesViewController"]
-        
-        self.tiledLayer = AGSArcGISTiledLayer(url: URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/World_Street_Map/MapServer")!)
-        let map = AGSMap(basemap: AGSBasemap(baseLayer: self.tiledLayer))
-        
-        self.mapView.map = map
-        
-        //add the graphics overlay to the map view
-        self.mapView.graphicsOverlays.add(self.graphicsOverlay)
-        
-        self.setupExtentView()
-        
-        self.previewMapView.layer.borderColor = UIColor.white.cgColor
-        self.previewMapView.layer.borderWidth = 8
+    /// A view to emphasize the extent of exported tile layer.
+    @IBOutlet var extentView: UIView! {
+        didSet {
+            extentView.layer.borderColor = UIColor.red.cgColor
+            extentView.layer.borderWidth = 2
+        }
     }
     
-    func setupExtentView() {
-        self.extentView.layer.borderColor = UIColor.red.cgColor
-        self.extentView.layer.borderWidth = 2
+    /// A view to provide a dark blurry background to preview the exported tiles.
+    @IBOutlet var visualEffectView: UIVisualEffectView!
+    /// A map view to preview the exported tiles.
+    @IBOutlet var previewMapView: AGSMapView! {
+        didSet {
+            previewMapView.layer.borderColor = UIColor.white.cgColor
+            previewMapView.layer.borderWidth = 8
+        }
+    }
+    /// A bar button to initiate or cancel the download task.
+    @IBOutlet var exportTilesBarButtonItem: UIBarButtonItem! {
+        didSet {
+            exportTilesBarButtonItem.possibleTitles = ["Export tiles", "Cancel"]
+        }
     }
     
-    func frameToExtent() -> AGSEnvelope {
-        let frame = self.mapView.convert(self.extentView.frame, from: self.view)
-        
-        let minPoint = self.mapView.screen(toLocation: frame.origin)
-        let maxPoint = self.mapView.screen(toLocation: CGPoint(x: frame.origin.x + frame.width, y: frame.origin.y + frame.height))
-        let extent = AGSEnvelope(min: minPoint, max: maxPoint)
-        return extent
+    // MARK: Properties
+    
+    /// The tiled layer created from world street map service.
+    let tiledLayer = AGSArcGISTiledLayer(url: URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/World_Street_Map/MapServer")!)
+    /// The export task to request the tile package with the same URL as the tile layer.
+    let exportTask = AGSExportTileCacheTask(url: URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/World_Street_Map/MapServer")!)
```

Comment:
Both share the same service URL. I tend to use a static let for the URL, but Phil might suggest against it. 🤔